src/wp-admin/includes/image.php

```diff
@@ -291,30 +291,55 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 		 * If the original image's dimensions are over the threshold,
 		 * scale the image and use it as the "full" size.
 		 */
+		$scale_down = false;
+		$convert    = false;
+
 		if ( $threshold && ( $image_meta['width'] > $threshold || $image_meta['height'] > $threshold ) ) {
+			// The image will be converted if needed on saving.
+			$scale_down = true;
+		} else {
+			// The image may need to be converted regardless of its dimensions.
+			$output_format = wp_get_image_editor_output_format( $file, $imagesize['mime'] );
+
+			if ( is_array( $output_format ) && array_key_exists( $imagesize['mime'], $output_format ) ) {
+				$convert = true;
+			}
+		}
+
+		if ( $scale_down || $convert ) {
 			$editor = wp_get_image_editor( $file );
 
 			if ( is_wp_error( $editor ) ) {
 				// This image cannot be edited.
 				return $image_meta;
 			}
 
-			// Resize the image.
-			$resized = $editor->resize( $threshold, $threshold );
+			if ( $scale_down ) {
+				// Resize the image. This will also convet it if needed.
+				$resized = $editor->resize( $threshold, $threshold );
+			} elseif ( $convert ) {
+				// The image will be converted (if possible) when saved.
+				$resized = true;
+			}
+
 			$rotated = null;
 
 			// If there is EXIF data, rotate according to EXIF Orientation.
 			if ( ! is_wp_error( $resized ) && is_array( $exif_meta ) ) {
 				$resized = $editor->maybe_exif_rotate();
-				$rotated = $resized;
+				$rotated = $resized; // bool true or WP_Error
```

Comment:
Is this comment allowed in WPCS? I always thought the rule of Capital letter at the beginning and a `.` at the end is strict?

Reply:
Hmm, not sure why this type of comments should be banned? Seems WPCS (the actual coding standards, and the PHPCS sniffs) don't have any restrictions for such inline comments. 

It may make sense to enforce capital letters and full stops when the comments are whole sentences (although that would be incompatible with the "Code is Poetry" motto, perhaps). This one is not a sentence :)

Reply:
I’ve checked the documentation and I can neither find anything that says it’s follows the coding standards nor that it does not. 
So I guess we can leave it like it is. 

https://developer.wordpress.org/coding-standards/inline-documentation-standards/php/#5-inline-comments